ygot/struct_validation_map.go

```diff
@@ -565,7 +597,11 @@ func copyStruct(dstVal, srcVal reflect.Value) error {
 			vSrc, vDst := srcField.Int(), dstField.Int()
 			switch {
 			case vSrc != 0 && vDst != 0 && vSrc != vDst:
-				return fmt.Errorf("destination and source values were set when merging enum field, dst: %d, src: %d", vSrc, vDst)
+				if !fieldOverwriteEnabled(opts) {
+					return fmt.Errorf("destination and source values were set when merging enum field, dst: %d, src: %d", vSrc, vDst)
+				} else {
```

Comment:
You don't need the `else` here, since the if the clause above is executed then an error is returned, so we can just have this as the `dstField.Set` outside of the `else`.